### PR TITLE
TOOLS/PERF: Multi-thread perftest enhancements

### PR DESCRIPTION
**Pull request by yosefe**

### CODE REVIEW

File: src/tools/perf/perftest.c [hunk 1]
```diff
@@ -721,9 +721,6 @@ static ucs_status_t parse_test_params(perftest_params_t *params, char opt,
         }
     case 'T':
         params->super.thread_count = atoi(opt_arg);
-        params->super.thread_mode  = (params->super.thread_count == 1) ?
```
Comment (dmitrygx) [inline src/tools/perf/perftest.c:724]: q: just wondering. does it mean that we will respect "-M" argument where thread mode is specified by the user?

Comment (yosefe) [reply src/tools/perf/perftest.c:724]: yes, thread mode is independent of # of threads, since each worker is on separate thread

Comment (alinask) [reply src/tools/perf/perftest.c:724]: could there be any perf impact if we run MULTI if only 1 thread is used? 
do we want to allow this case?

Comment (yosefe) [reply src/tools/perf/perftest.c:724]: yes, there will be extra lock overhead, and this is something we want to test as well

Comment (alinask) [reply src/tools/perf/perftest.c:724]: but isn't this considered a wrong usage? using Multi for 1 thread

Comment (yosefe) [reply src/tools/perf/perftest.c:724]: it's not optimal but i wouldn't say it's wrong

Comment (alinask) [reply src/tools/perf/perftest.c:724]: ok

**Comment by yosefe** [comment]
@alinask can you pls review as well?

**Comment by alinask** [comment]
@yosefe let's merge?

**Pull request merged**